AtTools/AtEDistortionModel.cxx

```diff
@@ -0,0 +1,191 @@
+   output.SetY(outputRad * TMath::Sin(outputAng));
+   output.SetZ(input.Z() - zcorr);
+
+   LOG(debug) << " Output point : " << output.X() << " " << output.Y() << " " << output.Z() << "\n";
+
+   return output;
+}
+
+XYZPoint AtEDistortionModel::ApplySpaceCharge(const XYZPoint &input)
+{
+   XYZPoint p;
+   return p;
+}
+
+Bool_t
+AtEDistortionModel::SetCorrectionMaps(const std::string &zlut, const std::string &radlut, const std::string &tralut)
+{
+
+   fZLUT.open(zlut, std::ifstream::in);
+   fRadLUT.open(radlut, std::ifstream::in);
+   fTraLUT.open(tralut, std::ifstream::in);
+
+   if (!fZLUT.fail() && !fRadLUT.fail() && !fTraLUT.fail()) {
+
+      SetBufferMap(fZLUT, fZMap.get());
+      SetBufferMap(fRadLUT, fRadMap.get());
+      SetBufferMap(fTraLUT, fTraMap.get());
+
+      SetGlobalMap(fZMap.get(), fRadMap.get(), fTraMap.get());
+
+      return kTRUE;
+
+   } else {
+
+      std::cerr << " AtEDistortionModel::SetCorrectionMaps - Error! Missing LUT "
+                << "\n";
+      std::cerr << zlut << "\n";
+      std::cerr << radlut << "\n";
+      std::cerr << tralut << "\n";
+      return kFALSE;
+   }
+
+   return kTRUE;
+}
+
+std::tuple<Double_t, Double_t, Double_t>
+AtEDistortionModel::GetCorrectionFactors(const Int_t &radius, const Int_t &zpos)
+{
+
+   if (fDistortionMap->size() == 0) {
+      LOG(error) << " Warning! Correction map is empty. "
+                 << "\n";
+      return std::make_tuple(0, 0, 0);
+   }
+
+   EFieldMapRef ref(radius, zpos);
+   if (fDistortionMap->find(ref) == fDistortionMap->end()) {
+      LOG(error) << " Correction factor not found. "
+                 << "\n";
+      return std::make_tuple(0, 0, 0);
+   } else {
+      auto corr = fDistortionMap->at(ref);
+      return std::make_tuple(corr.GetZCorr(), corr.GetRadCorr(), corr.GetTraCorr());
+   }
+}
+
+void AtEDistortionModel::SetBufferMap(std::ifstream &lut, BufferMap *map)
+{
+
+   while (!lut.eof()) {
+
+      std::string line;
+      std::getline(lut, line, '\r');
+      Int_t radius;
+      Double_t value;
+      std::istringstream buffer(line);
+      Int_t cnt = 0;
+
+      buffer >> radius;
+
+      while (buffer >> value) {
+
+         EFieldMapRef ref(radius, cnt);
+         map->emplace(ref, value);
+         ++cnt;
+      }
+   }
+}
+
+void AtEDistortionModel::SetGlobalMap(BufferMap *zmap, BufferMap *radmap, BufferMap *tramap)
+{
+
+   std::cout << " Z map size : " << zmap->size() << "\n";
+   std::cout << " Rad map size : " << radmap->size() << "\n";
+   std::cout << " Tra map size : " << tramap->size() << "\n";
+
+   // NB: This is not a necessary condition but it works for the present case
+   assert((zmap->size() == radmap->size()) && (radmap->size() == tramap->size()));
+
+   for (const auto &[key, zcorr] : *zmap) {
+      auto radcorr = radmap->at(key);
+      auto tracorr = tramap->at(key);
+      EFieldCorr corr(zcorr, radcorr, tracorr);
+      fDistortionMap->emplace(key, corr);
```

Comment:
Similar to above comment, can use aggregate initialization (https://en.cppreference.com/w/cpp/language/aggregate_initialization).